samples/wifi/promiscuous/src/main.c

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ * @brief Wi-Fi Promiscuous sample
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(promiscuous, CONFIG_LOG_DEFAULT_LEVEL);
+
+#ifdef CONFIG_NET_CONFIG_SETTINGS
+#include <zephyr/net/net_config.h>
+#endif
+
+#include <zephyr/net/ethernet.h>
+#include <zephyr/posix/unistd.h>
+#include <zephyr/posix/sys/socket.h>
+#include "wifi_connection.h"
+
+#define STACK_SIZE CONFIG_RX_THREAD_STACK_SIZE
+#define THREAD_PRIORITY K_PRIO_COOP(CONFIG_NUM_COOP_PRIORITIES - 1)
+#define RECV_BUFFER_SIZE 1024
+#define RAW_PKT_HDR 6
+
+/* Enumeration for Wi-Fi frame types and subtypes */
+enum frame_type {
+	MANAGEMENT_FRAME = 0x00,
+	CONTROL_FRAME = 0x01,
+	DATA_FRAME = 0x02,
+	RESERVED_FRAME = 0x03,
+	UNKNOWN_FRAME = 0xFF
+};
+
+enum control_subtype {
+	CF_END_SUBTYPE = 0x0E,
+	ACK_SUBTYPE = 0x0D,
+	RTS_SUBTYPE = 0x0B,
+	CTS_SUBTYPE = 0x0C,
+	BLOCK_ACK_SUBTYPE = 0x09,
+	BLOCK_ACK_REQ_SUBTYPE = 0x08
+};
+
+enum data_subtype {
+	DATA_SUBTYPE = 0x00,
+	QOS_DATA_SUBTYPE = 0x08,
+	NULL_SUBTYPE = 0x04,
+	QOS_NULL_SUBTYPE = 0x0C
+};
+
+enum management_subtype {
+	PROBE_REQ_SUBTYPE = 0x04,
+	BEACON_SUBTYPE = 0x08,
+	PROBE_RESPONSE_SUBTYPE = 0x05,
+	ACTION_SUBTYPE = 0x0D
+	/* Add more subtypes as needed */
+};
+
+struct packet_data {
+	int send_sock;
+	int recv_sock;
+	char recv_buffer[RECV_BUFFER_SIZE];
+};
+
+/* Structure for the MAC header */
+typedef struct {
+	uint16_t frame_control;
+	uint16_t duration;
+	uint8_t receiver_addr[6];
+	uint8_t transmitter_addr[6];
+	uint8_t bssid[6];
+	uint16_t sequence_control;
+	uint8_t address_4[6];
```

Comment:
This isn't part of standard header and only applicable if fromds+tods are set. And to catch such bugs its better to use some field from beacon body (fixes/IEs) and log a print.

Reply:
Hi Triveni,
I think you could just take the frame control field and provide a bitmap for the frame control elements. I think that is the only thing that is being checked in the code. I however do not believe that the above will cause a bug as it is never checked for the same in this code. But, a bitmap for frame control will be more concise as that is the only thing checked in the code.

Thanks and regards
Vivek